src/Filesystem/File.php

```diff
@@ -277,6 +277,15 @@ public function match(array $rules): bool
 		if (is_array($rules['mime'] ?? null) === true) {
 			$mime = $this->mime();
 
+			// the MIME type could not be determined, but matching
+			// to it was requested explicitly
+			if ($mime === null) {
+				throw new Exception([
```

Comment:
Could this be a NotFoundException for a bit more context?

Reply:
I copied over the pattern of the existing exceptions. Just doing it here would be inconsistent IMO and changing it everywhere could be a breaking change.

Reply:
If we want to use the more specific exception, would it be an option to use it already here and add todos for v5 to change the others? Then we reconcile consistency vs. breaking change

Reply:
Maybe it would make sense to introduce more specific exceptions that better fit the type of error instead of the current broad "catch-all" ones. For this error I can't find one that fully fits. `NotFoundException` would mean that the file itself is not found (which is one possible cause, but could also be that the file exists but has no detectable MIME type). `LogicException` would fit but is *very* broad.

Reply:
You are right. We could totally use better exceptions for various cases. But I guess this is way too much for this PR. So I would probably just stick with the Exception so far.